File: php/Scheme/Formula/Interval/AIntervalFormula.php
<?php

namespace Scheme\Formula\Interval;

use Scheme\Formula\Interval\IIntervalFormula;

abstract class AIntervalFormula implements IIntervalFormula {
	
  protected $distance = 0;	
  protected $sortingOrder = '>';

  //public function __construct() {}

  public function setDistance($distance) {
    $this->distance = $distance;
  }	

  public function getSortingOrder() {
    return $this->sortingOrder;
  }

  private function setSortingOrder($o) {
    $this->sortingOrder = $o;
  }

  public function setSortingOrderAsc() {
    $this->setSortingOrder('>');
  }

  public function setSortingOrderDesc() {
    $this->setSortingOrder('<');
  }
}

File: java/src/formula/interval/AIntervalFormula.java
package formula.interval;

public abstract class AIntervalFormula implements IIntervalFormula {
	
	protected int m_dt = 0;	
	protected char m_sortingOrder = '>';

	public AIntervalFormula() {		
		m_dt = 0;	
	}

	public void setDistance(int dt) {
		m_dt = dt;
	}	

	public char getSortingOrder() {
		return m_sortingOrder;
	}

	private void setSortingOrder(char o) {
		m_sortingOrder = o;
	}

	public void setSortingOrderAsc() {
		this.setSortingOrder('>');
	}

	public void setSortingOrderDesc() {
		this.setSortingOrder('<');
	}
}
